HZN-911: Add Geographical Status Map to Index Page

mvrueden opened the pull request:
* JIRA: https://issues.opennms.org/browse/HZN-911

mvrueden commented:
I found an issue: The heatmap-box is no longer working as it requires requirejs, but we do no longer include it on the index.jsp. I have to see what the other boxes such as grafana, trendline, etc. do.

pbrane commented:
I saw that, as well. I thought it was just me because I could get it to work, occasionally, if I constantly refreshed it like a mad jack rabbit on cocaine.

David 

> On Jan 26, 2017, at 09:22, mvrueden <notifications@github.com> wrote:
> 
> I found an issue: The heatmap-box is no longer working as it requires requirejs, but we do no longer include it on the index.jsp. I have to see what the other boxes such as grafana, trendline, etc. do.
>

mvrueden commented:
Should work now.

RangerRick commented:
Just checked out the branch and this looks pretty awesome. 👍

mvrueden commented on features/geolocation/feature/pom.xml:
```diff
@@ -0,0 +1,55 @@
+        </dependency>
+        <dependency>
+            <groupId>org.opennms</groupId>
+            <artifactId>opennms-dao-api</artifactId>
```
AFAIK this should be picked up via custom.properties and should not be part of the bundle/feature.

soleger replied:
Agreed.

j-white commented:
Looks good to me. Great work @mvrueden.

j-white commented on opennms-webapp/src/main/webapp/lib/bootstrap-sass-official/assets/stylesheets/bootstrap/_variables.scss:
```diff
@@ -355,7 +355,7 @@ $container-lg:                 $container-large-desktop !default;
 
 // Basics of a navbar
-$navbar-height:                    50px !default;
+$navbar-height:                    65px !default;
```
If we need to update this variable, we should set it on our theme instead of modifying it here.

j-white commented:
After compiling this branch, I'm left with a number of modified files in `opennms-webapp/src/main/webapp/lib`.

mvrueden commented:
I updated the pom dependencies, as mentioned above. In addition I reverted the navbar height changes, as they have side-effects e.g. at the topology map and it is not crucial. As soon as bamboo goes green, this can be merged, from my point of view. Please squash merge.

mvrueden commented:
@StCyr tested this as well and confirmed that it works.

christianpape commented on opennms-webapp/src/main/webapp/index.jsp:
```diff
@@ -63,7 +64,7 @@
 	<div class="col-md-6" id="index-contentmiddle">
 		<%
-			String centerUrl = System.getProperty("org.opennms.web.console.centerUrl", "/includes/categories-box.jsp");
+			String centerUrl = System.getProperty("org.opennms.web.console.centerUrl", "/includes/categories-box.jsp,/geomap/map-box.jsp");
```
Please also change the default in the opennms.properties file: "The default is the view of SLM/RTC categories: /includes/categories-box.jsp."

mvrueden replied:
Too late, already merged, please open an issue for this, if this is required.
